src/aot/aot.cpp

```diff
@@ -165,6 +172,59 @@ std::optional<std::vector<uint8_t>> generate_section(
   return out;
 }
 
+// Clones the shim to final destination while also injecting
+// the custom .btaot section.
+int build_binary(const std_filesystem::path &shim,
+                 const std::string &out,
+                 const std::vector<uint8_t> &section)
+{
+  std::error_code ec;
+  char cmd[1024];
+  int err = 1;
+  int n;
+
+  // Write out section data in a temporary file
+  std::ofstream secdata(AOT_SECDATA_TEMPFILE, std::ios_base::binary);
+  secdata.write(reinterpret_cast<const char *>(section.data()), section.size());
+  secdata.close();
+
+  // Respect user provided BPFTRACE_OBJCOPY if present
+  std::string objcopy = "objcopy";
+  if (auto c = std::getenv("BPFTRACE_OBJCOPY"))
+    objcopy = c;
+
+  // Resolve objcopy binary to full path
+  auto objcopy_full = find_in_path(objcopy);
+  if (!objcopy_full)
+  {
+    LOG(ERROR) << "Failed to find " << objcopy << " in $PATH";
+    goto out;
+  }
+
+  n = snprintf(cmd,
```

Comment:
nit: n is not a great variable name and doesn't seem to get re-used so why not inline this into the `if` check?

Reply:
Yeah good catch. I think we'll still want it to be a local var cuz `< 0` is error and `>= sizeof(cmd)` is also overflow error